code/datums/components/dripping.dm

```diff
@@ -47,8 +47,17 @@
 		qdel(src)
 
 
-/datum/component/dripping/proc/timed_drip()
+/datum/component/dripping/process()
 	var/mob/living/dripper = parent
+	if(QDELETED(dripper))
+		qdel(src)
+		return
```

Comment:
This doesn't look like a needed sanity check. Components are fairly decent at cleaning themselves up. If this case ever happens, this SHOULD runtime.